### DISCUSSION TITLE
FBA of Harvey and Harvetta

### DISCUSSION THREAD

**Discussion by Wieri9** [opencobra/cobrapy#1286] [Q&A] [answered]
>I tried to work with whole-body metabolic reconstructions (Harvey and Harvetta) but when I simulating flux balance analysis with Model.optimize() the funciontion don't give any results or error if I use 'GLPK', while 'Gurobi' and 'CPLEX' solvers return this error: 
![Screenshot_20220922_164723](https://user-images.githubusercontent.com/114151017/191779554-c26cbc82-55f4-4d1a-a0b2-a3618abe1806.png)

where is the problem?

**Comment by cdiener** [comment, marked as answer]
For CPLEX and Gurobi you will need a license. The versions that can be installed with pip are test versions with limits on the number of variables in the model. The commercial or academic licensees don't have that limitation. What errors are you getting with GLPK?

**Reply by Wieri9** [reply]
I installed Gurobi with the academy license and now works, with GLPK after 6 hours I stopped the run because the function usually need few second or minute.
Thank you for your advise

**Reply by cdiener** [reply]
That makes sense. Great to hear it works now!
